lib/src/components/NxList/__tests__/NxList.test.tsx

```diff
@@ -0,0 +1,232 @@
+      className: 'test-classname ufo'
+    };
+    const nxList = getShallow(extendedProps);
+    expect(nxList).toMatchSelector('.nx-list.test-classname.ufo');
+  });
+
+  it('renders a bulleted list correctly', function() {
+    const contentEl = getShallow({bulleted: true});
+    expect(contentEl).toHaveClassName('nx-list--bulleted');
+  });
+  it('shows the emptyMessage when there are no children', async function() {
+    const component = await mountAttached(
+      <NxList emptyMessage="Empty message">
+      </NxList>
+    );
+
+    expect(component.find('span')).toHaveText('Empty message');
+  });
+
+  it('shows the emptyMessage when the children are removed after the existing', async function() {
+    const component = await mountAttached(
+      <NxList emptyMessage="Empty message">
+        <NxList.Item>
+          <NxList.Text>Test 1</NxList.Text>
+        </NxList.Item>
+        <NxList.Item>
+          <NxList.Text>Test 2</NxList.Text>
+        </NxList.Item>
+      </NxList>
+    );
+
+    await act(async () => {
+      component.setProps({ children: [] });
+    });
+    component.update();
+
+    expect(component.find('span')).toHaveText('Empty message');
+  });
+
+  it('shows the emptyMessage when there are no children, no error, and not loading', async function() {
+
+    const trulyEmptyComponent = await mountAttached(<NxList emptyMessage="Empty message"></NxList>),
+        emptyListComponent = await mountAttached(<NxList emptyMessage="Empty message">{[]}</NxList>);
+
+    // the enzyme wrapper contains the <NxList> as its top element, the native el is one level down
+    expect(trulyEmptyComponent.children()).toMatchSelector('ul');
+    expect(trulyEmptyComponent).toContainExactlyOneMatchingElement('li.nx-list__item');
+    expect(trulyEmptyComponent.find('span')).toHaveText('Empty message');
+
+    expect(emptyListComponent.children()).toMatchSelector('ul');
+    expect(emptyListComponent).toContainExactlyOneMatchingElement('li.nx-list__item');
+    expect(emptyListComponent.find('span')).toHaveText('Empty message');
+  });
+
+  it('does not show the emptyMessage when there are children', async function() {
+    const component = await mountAttached(
+      <NxList emptyMessage="Empty message">
+        <NxList.Item>
+          <NxList.Text>Foo</NxList.Text>
+        </NxList.Item>
+      </NxList>
+    );
+
+    expect(component).not.toHaveText('Empty message');
+  });
+
+  it('does not show the emptyMessage when isLoading', async function() {
+    const component = await mountAttached(<NxList emptyMessage="Empty message" isLoading />);
+
+    expect(component).not.toHaveText('Empty message');
+  });
+
+  it('does not show the emptyMessage when in error', async function() {
+    const component = await mountAttached(
+      <NxList emptyMessage="Empty message" error="Errr" retryHandler={() => {}}>
+      </NxList>
+    );
+
+    expect(component).not.toHaveText('Empty message');
```

Comment:
Safer - `toHaveText` would pass if it included the empty message plus something else
```suggestion
    expect(component).not.toIncludeText('Empty message');
  });

  it('does not show the emptyMessage when isLoading', async function() {
    const component = await mountAttached(<NxList emptyMessage="Empty message" isLoading />);

    expect(component).not.toIncludeText('Empty message');
  });

  it('does not show the emptyMessage when in error', async function() {
    const component = await mountAttached(
      <NxList emptyMessage="Empty message" error="Errr" retryHandler={() => {}}>
      </NxList>
    );

    expect(component).not.toIncludeText('Empty message');
```